src/entity/base_entity.rs

```diff
@@ -745,10 +745,77 @@ pub trait EntityTrait: EntityName {
     fn delete_many() -> DeleteMany<Self> {
         Delete::many(Self::default())
     }
+
+    /// Delete a model based on primary key
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use sea_orm::{error::*, tests_cfg::*, *};
+    /// #
+    /// # #[smol_potat::main]
+    /// # #[cfg(feature = "mock")]
+    /// # pub async fn main() -> Result<(), DbErr> {
+    /// #
+    /// # let db = MockDatabase::new(DbBackend::Postgres)
+    /// #     .append_exec_results(vec![
+    /// #         MockExecResult {
+    /// #             last_insert_id: 0,
+    /// #             rows_affected: 1,
+    /// #         },
+    /// #     ])
+    /// #     .into_connection();
+    /// #
+    /// use sea_orm::{entity::*, query::*, tests_cfg::fruit};
+    ///
+    /// let delete_result = fruit::Entity::delete_by_id(1).exec(&db).await?;
+    ///
+    /// assert_eq!(delete_result.rows_affected, 1);
+    ///
+    /// assert_eq!(
+    ///     db.into_transaction_log(),
+    ///     vec![Transaction::from_sql_and_values(
+    ///         DbBackend::Postgres,
+    ///         r#"DELETE FROM "fruit" WHERE "fruit"."id" = $1"#,
+    ///         vec![1i32.into()]
+    ///     )]
+    /// );
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    fn delete_by_id(values: <Self::PrimaryKey as PrimaryKeyTrait>::ValueType) -> DeleteMany<Self> {
+        let mut delete = Self::delete_many();
+        let mut keys = Self::PrimaryKey::iter();
+        for v in values.into_value_tuple() {
+            if let Some(key) = keys.next() {
+                let col = key.into_column();
+                delete = delete.filter(col.eq(v));
+            } else {
+                panic!("primary key arity mismatch");
```

Comment:
Will this actually delete the code, I understand this might be inspired from `find_by_id` but I am not sure if this deletes the case

Reply:
The example code and the unit test work, but I'm not entirely sure.

Reply:
I'll tell why, find by ID  might be able to pull the data but delete is to remove data right? That kind of makes thing skeptical. I 'd also like to know if you can make it `delete_one` instead of `delete_many`

Reply:
Please use delete_one @ShouvikGhosh2048

Reply:
DeleteOne needs an a type that implements ActiveModelTrait. We're only passed an id.